contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3729,19 +3729,32 @@ exec_stmt_grantschema(PLtsql_execstate *estate, PLtsql_stmt_grantschema *stmt)
 		char	*rolname = NULL;
 		char	*grantee_name = (char *) lfirst(lc);
 		Oid	role_oid;
-		bool	grantee_is_db_owner = 0 == strcmp(grantee_name, db_owner);
 		bool	is_public = 0 == strcmp(grantee_name, PUBLIC_ROLE_NAME);
 		if (!is_public)
 			rolname	= get_physical_user_name(dbname, grantee_name);
 		else
 			rolname = pstrdup(PUBLIC_ROLE_NAME);
 		role_oid = get_role_oid(rolname, true);
 
-		if (!is_public && !OidIsValid(role_oid))
+		/* Special database roles should throw an error. */
+		if (strcmp(grantee_name, "db_owner") == 0)
 			ereport(ERROR, (errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
-				errmsg("Cannot find the principal '%s', because it does not exist or you do not have permission.", grantee_name)));
+				errmsg("Cannot grant, deny or revoke permissions to or from special roles.")));
+
+		if (!is_public && !OidIsValid(role_oid))
+		{
+			/* sys or information_schema roles should throw an error. */
+			if ((strcmp(grantee_name, "sys") == 0) || (strcmp(grantee_name, "information_schema") == 0))
```

Comment:
information_schema_tsql also?

Reply:
why?
`information_schema_tsql` is not a predefined SQL/database roles.

Reply:
that's for schema, grantee name is not rewritten to `information_schema_tsql`. Refer the catalog `sys.database_principals`